test/metabase/metabot/metabot_util_test.clj

```diff
@@ -155,26 +157,52 @@
     (mt/dataset sample-dataset
-      (let [{:keys [create_database_ddl]} (metabot-util/denormalize-database {:id (mt/id)})]
-        (is (str/includes? create_database_ddl "create type PRODUCTS_CATEGORY_t as enum 'Doohickey', 'Gadget', 'Gizmo', 'Widget';"))
-        (is (str/includes? create_database_ddl "create type PEOPLE_STATE_t as enum 'AK', 'AL', 'AR', 'AZ', 'CA', 'CO', 'CT',"))
-        (is (str/includes? create_database_ddl "create type PEOPLE_SOURCE_t as enum 'Affiliate', 'Facebook', 'Google', 'Organic', 'Twitter';"))
-        (is (str/includes? create_database_ddl "create type REVIEWS_RATING_t as enum '1', '2', '3', '4', '5';"))
-        (is (str/includes? create_database_ddl "CREATE TABLE \"PRODUCTS\" ("))
-        (is (str/includes? create_database_ddl "CREATE TABLE \"ORDERS\" ("))
-        (is (str/includes? create_database_ddl "CREATE TABLE \"PEOPLE\" ("))
-        (is (str/includes? create_database_ddl "CREATE TABLE \"REVIEWS\" ("))))))
+      (tu/with-temporary-setting-values [metabot-settings/enum-cardinality-threshold 50]
+        (let [{:keys [create_database_ddl]} (metabot-util/denormalize-database {:id (mt/id)})]
+          (is (str/includes? create_database_ddl "create type PRODUCTS_CATEGORY_t as enum 'Doohickey', 'Gadget', 'Gizmo', 'Widget';"))
+          (is (str/includes? create_database_ddl "create type PEOPLE_STATE_t as enum 'AK', 'AL', 'AR', 'AZ', 'CA', 'CO', 'CT',"))
+          (is (str/includes? create_database_ddl "create type PEOPLE_SOURCE_t as enum 'Affiliate', 'Facebook', 'Google', 'Organic', 'Twitter';"))
+          (is (str/includes? create_database_ddl "create type REVIEWS_RATING_t as enum '1', '2', '3', '4', '5';"))
+          (is (str/includes? create_database_ddl "CREATE TABLE \"PRODUCTS\" ("))
+          (is (str/includes? create_database_ddl "CREATE TABLE \"ORDERS\" ("))
+          (is (str/includes? create_database_ddl "CREATE TABLE \"PEOPLE\" ("))
+          (is (str/includes? create_database_ddl "CREATE TABLE \"REVIEWS\" (")))))))
+
+(deftest inner-query-test
+  (testing "Ensure that a dataset-based query contains AS aliases"
+    (mt/dataset sample-dataset
+      (t2.with-temp/with-temp
+        [Card orders-model {:name    "Orders Model"
+                            :dataset_query
+                            {:database (mt/id)
+                             :type     :query
+                             :query    {:source-table (mt/id :orders)}}
+                            :dataset true}]
+        (let [q (metabot-util/inner-query orders-model)
+              [_ aliases] (re-matches #"(?ism)SELECT(.+)FROM(:?.*)" q)]
+          (is (true? (str/includes? aliases " AS ")))))))
+  (testing "Ensure that a native dataset query contains a SELECT *"
+    (mt/dataset sample-dataset
+      (t2.with-temp/with-temp
+        [Card orders-model {:name          "Orders Model"
+                            :dataset_query {:database (mt/id)
```

Comment:
note your test here has a `dataset_query` so you're getting the aliases as `*` because of the `if-some` branch after doing a bunch of work in the function rather than the branch that is trying to recognize native queries and just quickly returning `*`